scilpy/denoise/asym_enhancement.py

```diff
@@ -59,18 +66,29 @@ def local_asym_gaussian_filtering(in_sh, sh_order=8, sh_basis='descoteaux07',
                             basis_type=sh_basis, return_inv=False,
                             full_basis=in_full_basis)
 
-    # Apply filter to each sphere vertice
-    for sf_i in range(nb_sf):
-        w_filter = weights[..., sf_i]
-
-        # Calculate contribution of center voxel
-        current_sf = np.dot(in_sh, B[:, sf_i])
-        mean_sf[..., sf_i] = w_filter[1, 1, 1] * current_sf
+    # Based on edge_mode generate a mask to separate background and foreground
+    if edge_mode == 'wall':
+        mask = in_sh[..., 0] > 0.
```

Comment:
Just tested this and I don't think this works as intended: voxels in SH volumes always have a bit of noise and so the mask just ends up covering the whole volume. I think passing an actual mask to the script would work better. Here's some files for comparison: https://drive.google.com/drive/folders/1H36FuGyf8oUTfq-Yp9d3Vu59zB-FSKkN?usp=sharing

I ran the script on `fibercup_fodf_8.nii.gz` with `sigma=2`, `sharpness=0.5` and `out_sym`. `no_wall.nii.gz` is without the `edge_mode` option. `wall.nii.gz` is with `--edge_mode=wall` and `masked.nii.gz` is where I modified the script to use a mask instead of computing it on line 71.

Reply:
I also used sigma=2 and sharpness=0.5 with the fibercup_fodf_8.nii.gz : https://drive.google.com/drive/folders/133OVcVVvxuXu7zfX2DXhMLJmfojPJbtc?usp=sharing

It's weird because it doesn't cover the whole volume in my case, and it seems like it's doing a good job of smoothing near the edges (fodf_8.nii.gz is the initial fodf, wall_sigma2.nii.gz using --edge_mode=wall, and no_wall_sigma2.nii.gz using default mode)

Reply:
@AntoineTheb
Right, it worked in my case because my input was TODI and background voxels were already clipped to 0. I'll add the option to specify a mask as argument when using `--edge_mode=wall` as you suggested. I might also add the option to specify a SH0 threshold manually.

@Florence-G
It seems to work on your dataset because it is already masked whereas @AntoineTheb's one is not. So for now if you just keep using masked input fODF I think it will work as intended.